db/migrate/20240311160949_add_product_to_licenses.rb

```diff
@@ -0,0 +1,14 @@
+class AddProductToLicenses < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    add_column :licenses, :product_id, :uuid, null: true
+
+    add_index :licenses, :product_id, algorithm: :concurrently
+
+    # FIXME(ezekg) Move to db/scripts?
+    Policy.find_each do |policy|
+      policy.licenses.update_all(product_id: policy.product_id)
```

Comment:
Move this out into a script. Need to think of how to make these types of data migrations easier for Keygen EE customers. Would be nice to run it inline here, but these types of migrations can take days for Keygen Cloud.

Reply:
I kind of love this: https://github.com/forem/forem/blob/main/app/models/data_update_script.rb#L1

Reply:
I like the idea of implementing `rails db:migrate:data`, which queues up a series of data migration jobs.

Reply:
What would be great is if we could somehow adjust the license's product association from a has-one through to a belongs-to based on whether or not existing data has been migrated or not. That way the data migration can be run async and it will eventually use the optimized denormalized column, and we don't have to require Keygen EE/CE users to run versions in a particular order.

Reply:
```rb
class License
  if exists?(product_id: nil)
    has_one :product, through: :policy
  else
    belongs_to :product
  end
end
```